### PR TITLE
Feature task2

### PR DESCRIPTION
**Pull request by vms-dev**

### CODE REVIEW

File: LookupSystem.DataAccess/Data/DbInitializer.cs [hunk 1]
```diff
@@ -1,9 +1,6 @@
-﻿using LookupSystemService.Models;
+﻿using LookupSystem.DataAccess.Models;
 using System;
```
Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Data/DbInitializer.cs:2]: Please review library name Bogus

File: LookupSystem.DataAccess/Models/User.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿using System;
 using System.ComponentModel.DataAnnotations.Schema;
 
-namespace LookupSystemService.Models
+namespace LookupSystem.DataAccess.Models
 {
     public class User
```
Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Models/User.cs:5]: As I know Datetime2 is by default please check

File: LookupSystem.DataAccess/Repositories/UserRepository.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                {
+                    db.Dispose();
+                }
+            }
+            this.disposed = true;
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
```
Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Repositories/UserRepository.cs:66]: What is it?

File: LookupSystem.DataAccess/Repositories/UserRepository.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
+﻿using LookupSystem.DataAccess.Data;
```
Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Repositories/UserRepository.cs:1]: Please review AsNoTracking and put where it needed

File: LookupSystemService/Controllers/UserController.cs [hunk 5]
```diff
@@ -12,19 +17,113 @@ namespace LookupSystemService.Controllers
...
             _logger = logger;
+            userRepo = new UserRepository(contex);
+        }
+
+        [HttpGet("GetUserByEmail/{email}")]
+        public IEnumerable<EmailSearchUserViewModel> GetUserByEmail(string email)
+        {
+            try
+            {
+                var config = new MapperConfiguration(cfg => cfg.CreateMap<User, EmailSearchUserViewModel>()
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Controllers/UserController.cs]: Is it correct place for mapper configuration? Please extract it in separate file

File: LookupSystemService/Controllers/UserController.cs [hunk 6]
```diff
@@ -12,19 +17,113 @@ namespace LookupSystemService.Controllers
...
+                .ForMember("FirstName", opt => opt.MapFrom(m => m.UserContact.FirstName))
+                .ForMember("LastName", opt => opt.MapFrom(m => m.UserContact.LastName))
+                .ForMember("Email", opt => opt.MapFrom(m => m.UserContact.Email)));
+
+                var mapper = new Mapper(config);
+                var users = mapper.Map<List<EmailSearchUserViewModel>>(userRepo.GetUserByEmail(email));
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Controllers/UserController.cs]: review automapper extension for EF and use it. Additional questions could be asked directly

File: LookupSystem.DataAccess/Repositories/UserRepository.cs [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                .ToList();
+
+            return user;
+        } 
+        
+        public IEnumerable<User> GetUserByPhone(string phone)
+        {
+            var user = db.Users                
+                .Include(c => c.UserContact)
+                .Where(u => u.UserContact.Phone.ToUpper() == phone.ToUpper())
```
Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Repositories/UserRepository.cs]: To Upper Produce new string, string.compare with ordinal ignore case no.

Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Repositories/UserRepository.cs]: Please  add index to phone column, and prove that you use it without conversion

File: LookupSystem.DataAccess/Repositories/UserRepository.cs [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        public IEnumerable<User> GetUserByEmail(string email)
+        {
+            var user = db.Users                
+                .Include(c => c.UserContact)
+                .Where(u => u.UserContact.Email.ToUpper() == email.ToUpper())
```
Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Repositories/UserRepository.cs]: same here.

File: LookupSystem.DataAccess/Models/UserContact.cs [hunk 9]
```diff
@@ -45,6 +45,6 @@ public class UserContact
```
Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Models/UserContact.cs:45]: Please change type for email to NVARCHAR

File: LookupSystem.DataAccess/Interfaces/IRepository.cs [hunk 10]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace LookupSystem.DataAccess.Interfaces
+{
+    public interface IRepository<T> : IDisposable  where T : class
```
Comment (OleksiiYatsenko) [inline LookupSystem.DataAccess/Interfaces/IRepository.cs:7]: Why we need repository ?

Comment (OleksiiYatsenko) [reply LookupSystem.DataAccess/Interfaces/IRepository.cs:7]: Which pattern developed inside of EF?

File: LookupSystemService/Controllers/UserController.cs [hunk 11]
```diff
@@ -12,19 +17,113 @@ namespace LookupSystemService.Controllers
...
+                return new List<EmailSearchUserViewModel>();
+            }
+        }
+
+        [HttpGet("GetUserByPhone/{phone}")]
+        public IEnumerable<PhoneSearchUserViewModel> GetUserByPhone(string phone)
+        {
+            try
+            {
+                var config = new MapperConfiguration(cfg => cfg.CreateMap<User, PhoneSearchUserViewModel>()
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Controllers/UserController.cs]: Same here for mapper

File: LookupSystemService/Models/EmailSearchUserViewModel.cs [hunk 12]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LookupSystemService.Models
+{
+    public class EmailSearchUserViewModel
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Models/EmailSearchUserViewModel.cs]: Do you have a view ? If no where you will take view model ?

Comment (OleksiiYatsenko) [reply LookupSystemService/Models/EmailSearchUserViewModel.cs]: Rename correctly for web API

File: LookupSystemService/Models/HiredUserViewModel.cs [hunk 13]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LookupSystemService.Models
+{
+    public class HiredUserViewModel
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Models/HiredUserViewModel.cs]: Same here

File: LookupSystemService/Models/PhoneSearchUserViewModel.cs [hunk 14]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LookupSystemService.Models
+{
+    public class PhoneSearchUserViewModel
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Models/PhoneSearchUserViewModel.cs]: Same Here

File: LookupSystemService/Models/UserViewModel.cs [hunk 15]
```diff
@@ -7,5 +7,20 @@ namespace LookupSystemService.Models
 {
     public class UserViewModel
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Models/UserViewModel.cs]: And here

File: LookupSystemService/Controllers/UserController.cs [hunk 16]
```diff
@@ -12,19 +17,113 @@ namespace LookupSystemService.Controllers
...
     {
+
         private readonly ILogger<UserController> _logger;
+        private readonly IRepository<User> userRepo;
 
-        public UserController(ILogger<UserController> logger)
+        public UserController(ILogger<UserController> logger, LookupSystemDbContext contex)
         {
             _logger = logger;
+            userRepo = new UserRepository(contex);
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Controllers/UserController.cs]: Use DI for it

File: LookupSystemService/Startup.cs [hunk 17]
```diff
@@ -39,6 +30,9 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddDbContext<LookupSystemDbContext>(options => options.UseSqlServer(Configuration.GetConnectionString("LookupSystemDbContext")));
```
Comment (OleksiiYatsenko) [inline LookupSystemService/Startup.cs:37]: Did you deploy SQL server ?

**Review by OleksiiYatsenko — COMMENTED** [review]
Do you ran this ??? Where is migrations.... Why you send it into review if here is absent db creation

File: LookupSystem.DataAccess/Repositories/UserRepository.cs (hunk 8 above)
Comment (vms-dev) [reply LookupSystem.DataAccess/Repositories/UserRepository.cs]: Error_1

При замене сравнения через оператор "==" на метод 'Compare' или 'Equals' я получаю ошибку при выполнении запроса. И как метод решения IDE предлагает использовать 'AsEnumerable' что по идее тоже не верно, потому как это буде выборка после выполнения запроса, или я ошибаюсь?

**Pull request merged**